src/main/java/org/threadly/util/debug/Profiler.java

```diff
@@ -231,6 +311,30 @@ public void run() {
             throw ExceptionUtils.makeRuntime(e.getCause());
           }
         }
+        // start or schedule to handle run time limit
+        if (sampleDurationInMillis > 0) {
+          if (executor instanceof SimpleSchedulerInterface) {
```

Comment:
We would have to do it in a 5.X since it would be a breaking API.  Here is my thoughts:

* If I were to re-do it now, I probably would not provide the executor option

* Though is adds complexity, as long as we are happy with the implementation, let's leave it for now.  The part I am unsure on is it is clear how we can possibly schedule on it, or the fact that it continually uses a thread from the executor, etc, etc.

So basically if we think the implementation is clean, let's keep it.  If it's kind of awkward or hard to understand, then it's probably worth a 3.9.0 release to remove it.